libs/brig-types/src/Brig/Types/User.hs

```diff
@@ -235,6 +235,45 @@ instance FromJSON SelfProfile where
 instance ToJSON SelfProfile where
     toJSON (SelfProfile u) = toJSON u
 
+----------------------------------------------------------------------------
+-- Rich info
+
+data RichInfo = RichInfo
+    { richInfoFields :: ![RichField]
+    }
+    deriving (Eq, Show)
+
+instance ToJSON RichInfo where
+    toJSON u = object
+        [ "fields" .= richInfoFields u
+        , "version" .= (0 :: Int)
+        ]
+
+instance FromJSON RichInfo where
+    parseJSON = withObject "RichInfo" $ \o -> do
+        o .:? "version" >>= \case
+            Nothing -> RichInfo <$> o .: "fields"
```

Comment:
I would (a) make the `version` field mandatory, or (b) do not introduce it at all now, but with the second version, and then make `Nothing` the first version.  Having two names (Nothing and Just 0) for one version seems confusing to me.

Reply:
I'll make it mandatory then.